### DISCUSSION TITLE
How to add name property to isDrawingMode drawn path?

### DISCUSSION THREAD

**Discussion by mortezasabihi** [fabricjs/fabric.js#7963] [Q&A] [answered]
Hi

I have a layers list in my application and when I create any shapes, a unique name will attach to the shape. This helps me to have the ability to select drawn objects on the canvas by clicking on the layer's item, like photoshop.
The problem is that I can't have the name property in PencilBrush or BaseBrush class.
![image](https://user-images.githubusercontent.com/35567646/169713365-e897cc32-d018-4e30-84ef-495bf152e750.png)

![image](https://user-images.githubusercontent.com/35567646/169713318-ef13113f-dbd6-470f-9c3b-c751380045c8.png)

**Comment by mortezasabihi** [comment]
I did this in this way:
![image](https://user-images.githubusercontent.com/35567646/169714072-876bb5b6-94f4-49e9-a807-3f9170f47c88.png)

But as you see there are ts issues but working.

**Comment by ShaMan123** [comment, marked as answer]
I know the docs are bad but your questions are very trivial and can be answered by taking a some time and reading the docs. 
http://fabricjs.com/fabric-intro-part-4#free_drawing
